api/controllers/ViewController.js

```diff
@@ -455,19 +458,180 @@ module.exports = {
+          if (thisEmail.validated) {
+            // The email is eligible to be primary.
+          } else {
+            reasons.push(`You selected ${thisEmail} to be your primary address, but it is not confirmed.`);
+          }
+        }
+      });
+    }
+
+    // No special validation needed for new emails at this time.
+    // If we wanted to validate, do it here.
+    if (request.payload.email_new) {}
+
+    // React to form validation errors.
+    if (reasons.length > 0) {
+      // Display the user feedback as an alert.
+      alert = {
+        type: 'danger',
+        message: '<p>Your profile could not be saved.</p><ul><li>' + reasons.join('</li><li>') + '</li></ul>',
+      };
+
+      // Show user the profile edit form again.
+      return reply.view('profile-edit', {
+        alert,
+        user,
+      });
+    } else {
+      // No errors were found, so load the user and update DB with the simple
+      // profile fields that don't need special treatment.
+      //
+      // TODO: replace this with UserController.update().
+      const user = await User.findOneAndUpdate({ _id: cookie.userId }, {
+        given_name: request.payload.given_name,
+        family_name: request.payload.family_name,
+      }, {
+        runValidators: true,
+        new: true,
+      });
+
+      logger.info(
+        `[ViewController->profileEditSubmit] Updated user profile for ${cookie.userId}`,
+        {
+          user: {
+            id: cookie.userId,
+          },
+        },
+      );
+
+      // Create a success confirmation.
+      cookie.alert = {
+        type: 'success',
+        message: '<p>Your profile was saved.</p>',
+      };
+
+      // Set primary email address using internal method. The first object is a
+      // mock of the request payload we'd send when making client-side JS calls.
+      if (request.payload.email_primary !== user.email) {
+        await UserController.setPrimaryEmail({}, {
+          userId: cookie.userId,
+          email: request.payload.email_primary,
+        }).then(data => {
+          cookie.alert.message += `<p>Your primary email was set to ${request.payload.email_primary}</p>`;
+        });
+      }
+
+      // If a new email address was submitted, add it to the user profile. The
+      // internal function will handle the confirmation email being sent.
+      if (request.payload.email_new) {
+        await UserController.addEmail({}, {
+          userId: cookie.userId,
+          email: request.payload.email_new,
+          appValidationUrl: _buildRequestUrl(request, 'verify2'),
+        }).then(data => {
+          cookie.alert.message += `<p>A confirmation email has been sent to ${request.payload.email_new}.</p>`;
+        }).catch(err => {
+          // Read our error and show some user feedback.
+          if (err.message.indexOf('Email is not unique') !== -1) {
+            cookie.alert.message += `<p>The address ${request.payload.email_new} is already added to your account.</p>`;
```

Comment:
@left23 per Flowdock discussion, this is where the error gets printed. but if we look deeper for logs that have a `[UserController->addEmail]` prefix, that would probably reveal the true cause for error.

Reply:
Yes it seems the `UserController.addEmail()` error doesn't differentiate between your account or someone else's. A simple change of wording when we handle the error will clear things up IMO.

> That email address has already been registered.